### PR TITLE
ref(issues): Strip relative path prefixes from stack frame title

### PR DESCRIPTION
**Pull request by Lms24**
As outlined in #50223, stack frame titles of errors from certain JS frameworks (e.g. SvelteKit or other projects built with Vite) can start with multiple `../` parent directory path prefixes.  This is confusing to our users, as they expect to see a filename that's relative to their project root. Therefore, this PR removes these parent directory hops from the frame title.

For more information/why I think this is best fixed in the UI, see #50223 and https://github.com/getsentry/sentry/issues/50223#issuecomment-1601093145

**Before**:
image

**After**:
image

### CODE REVIEW

File: static/app/components/events/interfaces/frame/defaultTitle/index.tsx [hunk 1]
```diff
@@ -42,6 +42,16 @@ function DefaultTitle({
...
   const tooltipDelay = isHoverPreviewed ? SLOW_TOOLTIP_DELAY : undefined;
 
+  /**
+   * In some cases (e.g. frame file names from SvelteKit stack frames),`frame.filename` can be
+   * prefixed with parent directory hops (e.g. `../../src/routes/....`).
+   * This function strips these hops so that users can clearly see where this file
+   * is located relative to their project root.
+   */
+  const stripRelativePathPrefix = (filename: string): string => {
+    return filename.replace(/^(..\/)*/, '');
```
Comment (scttcper) [inline static/app/components/events/interfaces/frame/defaultTitle/index.tsx:52]: maybe escape the periods?
```suggestion
    return filename.replace(/^(\.\.\/)*/, '');
```

Comment (Lms24) [reply static/app/components/events/interfaces/frame/defaultTitle/index.tsx:52]: oof good catch, totally missed this, thanks :D

**Review by scttcper — APPROVED** [review]
i'm not sure if this will be confusing for people setting up code mappings

**Review by mydea — APPROVED** [review]

**Comment by Lms24** [comment]
> setting up code mappings

@scttcper I'm not very familiar with Sentry UI specifics, so to clarify: Do you mean the "Tell us where your source code is" dialog of the GitHub integration? Do we have this for other integrations as well?

**Comment by scttcper** [comment]
@Lms24 yeah they tell us where your source code is thing for gitlab & github. Basically they need to set it up from issue details which is easy or in the integration itself and they have to map the path of the stack to the path in the repo. 

You can see them here https://sentry.sentry.io/settings/integrations/github/2933/ 
image

docs https://docs.sentry.io/product/integrations/source-code-mgmt/github/#stack-trace-linking

**Pull request closed without merging**